### DISCUSSION TITLE
How to change source file? (task plugin integration)

### DISCUSSION THREAD

**Discussion by cacosat** [ivan-lednev/obsidian-day-planner#490] [Q&A]
Hello, I'm trying to change the mode of the plugin so i can see my tasks made with the tasks plugin. I added the #task tag in the "Dataview Source" tab within the dayplanner timeline in the left side panel and also in another file where i have my tasks, but it didn't work. I've tried a few things but i can't seem to get it working, would apreciate some guidance.

**Comment by claremacrae** [comment]
Hi, I was just able to set the Dataview source to:

```
"/"
```

(I don't use a Tasks global filter, hence not showing a tag there...)

And it does find Tasks plugin tasks throughout my vault...
With the big caveat that, as per the docs, it only shows tasks with ⏳- scheduled.
And not 📅 - due (it wouldn't show 🛫 - starts either)

**Reply by ivan-lednev** [reply]
Hello, @claremacrae, I've been thinking about adding support for the other props, and Dataview cache will allow for that (at least `start` and `due`), but first I'll need to make it possible to add time to each of those props separately (something like `⏳ 2024-01-01 12:30 - 13:30`). With this, I'll be able to show a time block per each property, and the user will be able to edit each property separately with drag-and-drop. I'm aware that there is an issue in your repo that's asking exactly for that.

I considered adding extra props to track time separately (like `[tasks-due-time::12:00]`), but it seemed too hacky 🙈

**Reply by claremacrae** [reply]
Hi @ivan-lednev, many thanks for the reply.

But .... Please don't start adding times to the Tasks date signifiers before Tasks does.

It would create a support nightmare for me, with people complaining that Tasks is not reading their dates any more.

I've implemented something much simpler in my fork - I apply the time to one of the dates on the task... the only personal preference is the priority to use, when selecting which date to use:

https://github.com/claremacrae/obsidian-day-planner/compare/5ae2272febc5e643515a70588037f4a9fd4def0b...claremacrae:obsidian-day-planner:3f50b9044e1a6ee8df1d3bdf52bf15e84e088be4

I know this isn't releasable, but the point is that it is a much more localised change, that will continue to play nicely with Dataview and Tasks formats.
